[Bug]: TDZ ReferenceError when Storyblok components have circular dependency through `StoryblokComponent`

### Package

@storyblok/astro

### Bug Description

When two registered Storyblok components form a circular dependency through `StoryblokComponent.astro`, Vite 6's SSR bundler emits them into a single chunk with deferred getters. Because the `virtual:import-storyblok-components` virtual module eagerly accesses `.default` on every glob'd module at module initialization time, the deferred getter fires before the target variable is initialized, producing a TDZ `ReferenceError`.

The cycle is: `Card.astro` imports `Richtext.astro`, which imports `StoryblokComponent.astro` (from `@storyblok/astro`), which imports `virtual:import-storyblok-components`, which eagerly re-imports `Card.astro` via the `import.meta.glob(..., { eager: true })` call and immediately reads `modules[filePath].default` at the top level.

### Steps to Reproduce

1. Create two Storyblok components where one imports the other, and the second imports `StoryblokComponent` from `@storyblok/astro`:

   ```
   Card.astro → imports Richtext.astro
   Richtext.astro → imports @storyblok/astro/StoryblokComponent.astro
   StoryblokComponent.astro → imports virtual:import-storyblok-components
   virtual module → eagerly imports Card.astro (cycle)
   ```

2. Register both in `astro.config.ts`:

   ```ts
   storyblok({
     accessToken: "dummy",
     components: {
       card: "storyblok/Card",
       text: "storyblok/Richtext",
     },
   })
   ```

3. Create a page that renders `Card`.

4. Run `astro build`.

### Reproduction URL

https://stackblitz.com/github/adamchal/storyblok-tdz?startScript=build

### Expected Behavior

The build completes successfully. Circular component dependencies are a natural pattern in CMS-driven sites (a Card can contain rich text, and rich text can contain embedded Storyblok blocks via `StoryblokComponent`).

### Actual Behavior

The build fails during static generation with:

```
ReferenceError: Cannot access '$$Card' before initialization
```

### Environment

```shell
System:
    OS: macOS 15.7.4
    CPU: (16) x64 Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
  Binaries:
    Node: 25.8.2
    npm: 11.12.1
  npmPackages:
    @storyblok/astro: ^9.0.3 => 9.0.3
    astro: ^6.1.3 => 6.1.3
```

### Error Logs

```shell
14:46:08 [ERROR] ReferenceError: Cannot access '$$Card' before initialization
    at get default (file:///tmp/storyblok-tdz/dist/.prerender/chunks/index_Cj6juoCk.mjs:12:20)
    at file:///tmp/storyblok-tdz/dist/.prerender/chunks/index_Cj6juoCk.mjs:35435:45
    at ModuleJob.run (node:internal/modules/esm/module_job:430:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:639:26)
    at async BuildPipeline.getModuleForRoute
```

### Additional Context

The root cause is in the generated `virtual:import-storyblok-components` module (`vite-plugin-import-storyblok-components`). It does:

```js
const modules = import.meta.glob('.../storyblok/**/*.astro', { eager: true });
const storyblokComponents = {};
for (const filePath in modules) {
  const component = modules[filePath].default || modules[filePath]; // <-- TDZ here
  storyblokComponents[camelCaseName] = component;
}
```

A fix would be to defer the `.default` access until render time, e.g. using lazy property getters:

```js
const modules = import.meta.glob('.../storyblok/**/*.astro', { eager: true });
const storyblokComponents = {};
for (const filePath in modules) {
  const camelCaseName = toCamelCase(componentName);
  Object.defineProperty(storyblokComponents, camelCaseName, {
    get() { return modules[filePath].default || modules[filePath]; },
    enumerable: true,
    configurable: true,
  });
}
```

This defers `.default` access to render time, when all module-level variables are fully initialized.

--- a/packages/astro/src/components/StoryblokComponent.astro
+++ b/packages/astro/src/components/StoryblokComponent.astro
@@ -11,34 +11,25 @@ export interface Props {
 const { blok, ...props } = Astro.props;
 
 if (!blok) {
-  throw new Error(
-    "Cannot render StoryblokComponent. 'blok' prop is undefined."
-  );
+  throw new Error("Cannot render StoryblokComponent. 'blok' prop is undefined.");
 }
 /**
  * convert blok components to camel case to match vite-plugin-import-storyblok-components
  */
 let key = toCamelCase(blok.component as string);
 
-const componentFound: boolean = key in storyblokComponents;
+const loader = storyblokComponents[key] || storyblokComponents['FallbackComponent'];
 
-let Component: AstroComponentFactory;
-
-if (!componentFound) {
-  if (!options.enableFallbackComponent)
-    throw new Error(
-      `No component found for blok "${blok.component}". 
+if (!loader) {
+  throw new Error(
+    `No component found for blok "${blok.component}". 
       Make sure the component is:
       • Registered in your astro.config.mjs, or
       • Placed in the "/${options.componentsDir}/storyblok" folder, or
-      • Enable the "fallbackComponent" option to handle missing components.`
-    );
-  else {
-    Component = storyblokComponents['FallbackComponent'];
-  }
-} else {
-  Component = storyblokComponents[key];
+      • Enable the "fallbackComponent" option to handle missing components.`,
+  );
 }
+const Component: AstroComponentFactory = await loader();
 ---
 
 <Component blok={blok} {...props} />

--- a/packages/astro/src/internal.d.ts
+++ b/packages/astro/src/internal.d.ts
@@ -21,7 +21,10 @@ declare namespace App {
 declare module 'virtual:import-storyblok-components' {
   import type { AstroComponentFactory } from 'astro/runtime/server/index.js';
 
-  export const storyblokComponents: Record<string, AstroComponentFactory>;
+  export const storyblokComponents: Record<
+    string,
+    () => Promise<AstroComponentFactory>
+  >;
 }
 
 /** Integration options provided to the Astro SDK. */

--- a/packages/astro/src/vite-plugins/vite-plugin-import-storyblok-components.ts
+++ b/packages/astro/src/vite-plugins/vite-plugin-import-storyblok-components.ts
@@ -90,40 +90,31 @@ function generateModuleCode(
   return `
     // Import utilities and fallback component
     import { toCamelCase } from '@storyblok/astro';
-    ${fallbackImport}
-    ${manualImports.join('\n')}
 
     // Dynamically import all Storyblok components using Vite's glob import
-    const modules = import.meta.glob('${globPattern}', { eager: true });
+    const modules = import.meta.glob('${globPattern}');
     // Process imported modules into a components object
     const storyblokComponents = {};
-    
     for (const filePath in modules) {
       // Extract component name from file path (remove extension)
       const fileName = filePath.split('/').pop();
       const componentName = fileName?.replace(/\\.[^/.]+$/, '');
       
       if (componentName) {
-        // Get the component (handle both default and named exports)
-        const component = modules[filePath].default || modules[filePath];
-        
         // Convert filename to camelCase for Storyblok component naming
         const camelCaseName = toCamelCase(componentName);
-        storyblokComponents[camelCaseName] = component;
+
+        storyblokComponents[camelCaseName] = async () => {
+        const module = await modules[filePath]();
+        return module.default || module;
+        };
       }
     }
     
-    // Manual imports (overwrite auto if same key exists)
-    ${manualImports
-      .map((imp) => {
-        const varName = imp.match(/import\s+(\w+)\s+from/)?.[1];
-        return `storyblokComponents['${varName}'] = ${varName};`;
-      })
-      .join('\n')}
-
+    // Manual components
+    ${manualImports.join('\n')}
     // Add fallback component if enabled
-    ${fallbackImport ? `storyblokComponents['FallbackComponent'] = FallbackComponent;` : ''}
-    
+    ${fallbackImport}    
     // Export the components object for use in Storyblok initialization
     export { storyblokComponents };
   `.trim();
@@ -150,10 +141,19 @@ async function resolveFallbackComponent(
         `Custom fallback component could not be found. Does "${customPath}" exist?`,
       );
     }
-    return `import FallbackComponent from '${customPath}';`;
+
+    return `
+      storyblokComponents['FallbackComponent'] = async () => {
+        const module = await import('${resolved.id}');
+        return module.default || module;
+      };`;
   }
 
-  return `import FallbackComponent from '@storyblok/astro/FallbackComponent.astro';`;
+  return `
+    storyblokComponents['FallbackComponent'] = async () => {
+      const module = await import('@storyblok/astro/FallbackComponent.astro');
+      return module.default || module;
+    };`;
 }
 /**
  * Resolves user-provided Storyblok components into import statements.
@@ -184,7 +184,13 @@ async function resolveUserComponents(
       }
     }
     else {
-      imports.push(`import ${toCamelCase(key)} from "${resolvedId.id}"`);
+      const camelCaseName = toCamelCase(key);
+
+      imports.push(`
+      storyblokComponents['${camelCaseName}'] = async () => {
+        const module = await import('${resolvedId.id}');
+        return module.default || module;
+      };`);
     }
   }
 

--- a/packages/astro/src/vite-plugins/vite-plugin-import-storyblok-components.ts
+++ b/packages/astro/src/vite-plugins/vite-plugin-import-storyblok-components.ts
@@ -99,14 +99,14 @@ function generateModuleCode(
       // Extract component name from file path (remove extension)
       const fileName = filePath.split('/').pop();
       const componentName = fileName?.replace(/\\.[^/.]+$/, '');
-      
-      if (componentName) {
-        // Convert filename to camelCase for Storyblok component naming
-        const camelCaseName = toCamelCase(componentName);
 
-        storyblokComponents[camelCaseName] = async () => {
-        const module = await modules[filePath]();
-        return module.default || module;
+        if (componentName) {
+          // Convert filename to camelCase for Storyblok component naming
+          const camelCaseName = toCamelCase(componentName);
+
+          storyblokComponents[camelCaseName] = async () => {
+          const module = await modules[filePath]();
+          return module.default || module;
         };
       }
     }

